receptai.api/Repositories/RecipeRepository.cs

```diff
@@ -38,28 +38,60 @@ public async Task<Recipe> CreateAsync(Recipe recipeModel)
         return recipeModel;
     }
 
-    public async Task<List<Recipe>> GetAllAsync()
+    public async Task<List<Recipe>> GetAllAsync(int pageNumber = 1, int pageSize = int.MaxValue)
     {
-        return await _context.Recipes.ToListAsync();
+        if (pageNumber < 1) throw new ArgumentOutOfRangeException(nameof(pageNumber), "Page number should be greater than or equal to 1.");
+        if (pageSize < 1) throw new ArgumentOutOfRangeException(nameof(pageSize), "Page size should be greater than or equal to 1.");
+
+        var query = _context.Recipes.AsQueryable();
+
+        if (pageNumber > 1 || pageSize < int.MaxValue)
+        {
+            query = query
+                .Skip((pageNumber - 1) * pageSize)
+                .Take(pageSize);
+        }
+
+        return await query.ToListAsync();
```

Comment:
IIRC we agreed that concept of "page" shouldn't be in backend 😄

IMO it's better to explicitly say that it's offset from Y amount, and we want X amount of posts. Because it allows for more flexibility & non-uniform size requests (i.e. maybe first we fetch 5 then 50)

Reply:
Fixed.